localstack-core/localstack/services/apigateway/resource_providers/aws_apigateway_resource.py

```diff
@@ -94,6 +96,30 @@ def read(
         """
         raise NotImplementedError
 
+    def list(
+        self,
+        request: ResourceRequest[ApiGatewayResourceProperties],
+    ) -> ProgressEvent[ApiGatewayResourceProperties]:
+        input_payload = json.loads(request.desired_state)
```

Comment:
@simonrw I was playing with the resource endpoint and I came across this.
This is the only `list` operation so far that requires a desired state to be provided by `cloudcontrol`. When this does not happen, we pass an empty dictionary here, which of course fails to be JSON-loaded. From the rest of the code it looks like we could bound `Properties` to `dict`.

Reply:
What we should do to be in parity here is raise an exception if the required payload is not provided. I get the error message:

```
botocore.errorfactory.InvalidRequestException: An error occurred (InvalidRequestException) when calling the ListResources operation: Missing or invalid ResourceModel property in AWS::ApiGateway::Resource list handler request input.Required property:  (#: required key [RestApiId] not found)
```

I have just pushed a change that should make input handling better.